### DISCUSSION TITLE
Drumkit variants

### DISCUSSION THREAD

**Discussion by github-wlk** [hydrogen-music/hydrogen#1134] [Ideas]
Hi, I use hydrogen on Linux, and here is an idea that I would like to see.

Drumkit variants. New "drumkit" but based on a more "fixed" set of source files (wav), in order to save space and make things more flexible. So, let's say you have drumkit X, and you want to save a version of it which is using a different midi channel. It could be a different drumkit.xml file, or a different stanza in the "standard" kit's drumkit.xml file. I'll let you argue the merits of those options. I would just like to see more flexibility in having various versions and tweaks of a certain drumkit, without having the overhead of saving a whole new copy of the wavs.

Thanks!

**Comment by theGreatWhiteShark** [comment]
Hey @github-wlk ,

This is already what a song does provide.

A drumkit consists of samples and a default configuration of all parameters in the `drumkit.xml`. By loading a drumkit into a song you also load the default parameters. Changing them and saving multiple versions of the song (in `.h2song` files) does give you different versions of the drumkit parameters while all of them use the same samples.

**Reply by github-wlk** [reply]
Cool, I hadn't made that connection.  I'll have to move forward using that info.  Off the top of my head, a problem is that one can't apply those changes to another song, no?  One has song X, with a bunch of patterns, and song Y, with a bunch of patterns, and one can't say "apply drumkit from song X to song Y".   Yes, one should start out with the right tuning :)   So, the way I'll try to work this, is create hive of song "templates", each with its unique drumkit attributes, and then use that as the launching point for new tunes.

I guess we could share these songs as templates (perhaps as an unofficial convention have "template" in the name).  Just an idea :)

Thanks for the feedback, and this does help.

**Reply by oddtime** [reply]
I like this idea,
exporting the drumkit file without the samples

**Reply by theGreatWhiteShark** [reply]
> Cool, I hadn't made that connection. I'll have to move forward using that info. Off the top of my head, a problem is that one can't apply those changes to another song, no? One has song X, with a bunch of patterns, and song Y, with a bunch of patterns, and one can't say "apply drumkit from song X to song Y". Yes, one should start out with the right tuning :) So, the way I'll try to work this, is create hive of song "templates", each with its unique drumkit attributes, and then use that as the launching point for new tunes.

I also faced this a couple of times when programming different songs for an album and trying to align their settings. If you like, you can make a feature request https://github.com/hydrogen-music/hydrogen/issues. Maybe there will be an intuitive solution some day

**Reply by github-wlk** [reply]
Created https://github.com/hydrogen-music/hydrogen/issues/1157 ....   oddtime and tGWS, tear it apart :)

**Reply by charbeljc** [reply]
Hum, in fact, song provides mostly this, but UX is poor, at least. I think we should save a preset.h2song inside each drumkit directory and load/apply/save it each time we load/save the given drumkit. Maybe later we could extend it with many presets to choose for each drumkit.
